### PR TITLE
fix(sql): fix results when combining negative limit, column aliasing and predicate

### PR DESCRIPTION
**Pull request by jerrinot**
Consider this query: `select ts, symbol as id from t1 where symbol = 'abc' limit -1`
without this patch the plan will be similar to this:
```
Limit lo: -1
    SelectedRecord
        Async JIT Filter
          limit: 1
          pageFrameDirection: backward
          filter: symbol='abc'
          workers: 14
            DataFrame
                Row forward scan
                Frame forward scan on: t1
```
As you can see the limit is applied twice: once in `Async JIT Filter` and then on the top level once again. 
Why? The `AsyncJitFilteredRecordCursorFactory.followedLimitAdvice()` returns true. So normally `SqlCodeGenerator.generateLimit()` would not wrap it in additional Limit factory. But when `AsyncJitFilteredRecordCursorFactory` is wrapped in `SelectedRecordCursorFactory` (for example when aliasing columns), then `SelectedRecordCursorFactory` won't propagate the `followedLimitAdvice()` from the `AsyncJitFilteredRecordCursorFactory` and `SqlCodeGenerator` add additional limit on its own. 

This changeset simply makes `SelectedRecordCursorFactory` to propagate the `followedLimitAdvice()` flag of the underlayed RecordCursorFactory. So SqlCodeGenerator won't add additional limit.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java|2|2|100.00%|

**Comment by ideoma** [comment]
Can you add the description please e.g. the query you're fixing?

**Comment by jerrinot** [comment]
@ideoma: added a description to this PR.

**Review by ideoma — DISMISSED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java|2|2|100.00%|

File: core/src/test/java/io/questdb/griffin/LimitTest.java [hunk 1]
```diff
@@ -224,6 +224,24 @@ public void testLastN() throws Exception {
         testLimit(expected, expected2, query);
     }
 
+    @Test
+    public void testLimitMinusOneAndPredicateAndColumnAlias() throws Exception {
+        compiler.compile("create table t1 (ts timestamp, id symbol)", sqlExecutionContext);
+
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/griffin/LimitTest.java:230]: Not an issue but you can use insert into with multiple rows to remove repetition : 

```sql
insert into t1 values (0L, 'abc'), (2L, 'a1'), ... 
```

File: core/src/test/java/io/questdb/griffin/LimitTest.java [hunk 2]
```diff
@@ -224,6 +224,24 @@ public void testLastN() throws Exception {
         testLimit(expected, expected2, query);
     }
 
+    @Test
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/griffin/LimitTest.java:227]: Test content should be wrapped in assertMemoryLeak().

File: core/src/test/java/io/questdb/griffin/LimitTest.java (hunk 1 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/griffin/LimitTest.java:230]: thanks. to was a copy'n'paste from some existing test. let me change that.

File: core/src/test/java/io/questdb/griffin/LimitTest.java (hunk 2 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/griffin/LimitTest.java:227]: good catch! thanks!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java|2|2|100.00%|

**Review by bziobrowski — APPROVED** [review]
Looks good !

**Pull request merged**